modules/tftp/server.rb

```diff
@@ -7,33 +7,35 @@ class Server
     # Creates TFTP pxeconfig file
     def set mac, config
       raise "Invalid parameters received" if mac.nil? || config.nil?
-
       FileUtils.mkdir_p pxeconfig_dir
-
-      File.open(pxeconfig_file(mac), 'w') {|f| f.write(config) }
-      logger.info "TFTP: entry for #{mac} created successfully"
+      pxeconfig_file(mac).each do |file|
+        File.open(file, 'w') {|f| f.write(config) }
+        logger.info "TFTP: #{file} created successfully"
+      end
     end
 
     # Removes pxeconfig files
     def del mac
-      file = pxeconfig_file(mac)
-      if File.exist?(file)
-        FileUtils.rm_f file
-        logger.debug "TFTP: entry for #{mac} removed successfully"
-      else
-        logger.info "TFTP: Skipping a request to delete a file which doesn't exists"
+      pxeconfig_file(mac).each do |file|
+        if File.exist?(file)
+          FileUtils.rm_f file
+          logger.info "TFTP: #{file} removed successfully"
+        else
+          logger.warn "TFTP: Skipping a request to delete a file which doesn't exists"
```

Comment:
> > +          logger.warn "TFTP: Skipping a request to delete a file which doesn't exists"
> 
> This one too?

This one is actually correct I think - this is a reverse of previous
orchestration and it should always work correctly. We should definitely
leave a warning message when there was an inconsistency (user e.g. added
TFTP association for previously provisioned host).

-- 
Later,
 Lukas #lzap Zapletal